agent/doc/agent-doc.md

```diff
@@ -0,0 +1,24 @@
+# Centreon Agent documentation {#mainpage}
+
+## Introduction
+
+The goal of this program is to execute checks in both windows and linux OS
+It's full asynchronous, excepted grpc layers, it's single threaded and you won't find mutex in non grpc code.
+This is why when we receive request, we post it to asio in order to process it in the main thread.
+
+## Configuration
+configuration is given by Engine by a AgentConfiguration sent over grpc
+The configuration object is embedded in MessageToAgent::config
+
+## Scheduler
+We trie to spread checks over check_period.
+Example: We have 10 checks to execute during one second. check1 will start at now + 0.1s, second at now + 0.2s..
+
+When Agent receives configuration, all checks are recreated.
+For example, we have 100 checks to execute in 10 minute, at it is 12:00:00.
+First  service check will start right now, second one at 12:00:06, third at 12:00:12... and the last one at 12:09:54
```

Comment:
I have the feeling there is a contradiction here.
In the first example, you say check1 is done at now + 0.1s whereas in the second example, the first check seems to be now,

Reply:
done